monitoring/backups_collector.go

```diff
@@ -0,0 +1,98 @@
+package monitoring
+
+import (
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/longhorn/longhorn-manager/datastore"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+)
+
+type BackupsCollector struct {
+	*baseCollector
+
+	sizeMetric  metricInfo
+	stateMetric metricInfo
+}
+
+func NewBackupsCollector(
+	logger logrus.FieldLogger,
+	nodeID string,
+	ds *datastore.DataStore) *BackupsCollector {
+
+	vc := &BackupsCollector{
+		baseCollector: newBaseCollector(subsystemBackups, logger, nodeID, ds),
+	}
+
+	vc.sizeMetric = metricInfo{
+		Desc: prometheus.NewDesc(
+			prometheus.BuildFQName(longhornName, subsystemBackups, "actual_size_bytes"),
+			"Actual size of this backup",
+			[]string{volumeLabel, backupsLabel},
+			nil,
+		),
+		Type: prometheus.GaugeValue,
+	}
+
+	vc.stateMetric = metricInfo{
+		Desc: prometheus.NewDesc(
+			prometheus.BuildFQName(longhornName, subsystemBackups, "state"),
+			"State of this backup",
+			[]string{volumeLabel, backupsLabel},
+			nil,
+		),
+		Type: prometheus.GaugeValue,
+	}
+
+	return vc
+}
+
+func (vc *BackupsCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- vc.sizeMetric.Desc
+	ch <- vc.stateMetric.Desc
+}
+
+func (vc *BackupsCollector) Collect(ch chan<- prometheus.Metric) {
+	defer func() {
+		if err := recover(); err != nil {
+			vc.logger.WithField("error", err).Warn("panic during collecting metrics")
+		}
+	}()
+
+	backupsLists, err := vc.ds.ListBackupsRO()
+	if err != nil {
+		vc.logger.WithError(err).Warn("error during scrape ")
+		return
+	}
+
+	for _, v := range backupsLists {
+		if v.Status.OwnerID == vc.currentNodeID {
+			var size float64
+			if size, err = strconv.ParseFloat(v.Status.Size, 64); err != nil {
+				vc.logger.WithError(err).Warn("error get size")
+			}
+			ch <- prometheus.MustNewConstMetric(vc.sizeMetric.Desc, vc.sizeMetric.Type, size, v.Status.VolumeName, v.Name)
+			ch <- prometheus.MustNewConstMetric(vc.stateMetric.Desc, vc.stateMetric.Type, float64(getBackupsStateValue(v)), v.Status.VolumeName, v.Name)
```

Comment:
The actual volume name is stored by label, you could refer to this [code](https://github.com/longhorn/longhorn-manager/blob/436bda78a955bdaf4bd9a73bc5fd4613f84bcb24/controller/backup_controller.go#L401-L407).
Using v.Status.VolumeName might be empty because the status would be updated once it's synced/pulled from the remote backup target.

Reply:
Resolved on new commit b2145965748e3ec8da25eec1af7c15c48aab3d68